analyzer_oneapi.cc

```diff
@@ -0,0 +1,119 @@
+#include <chrono>
+#include <cstring>
+#include <iostream>
+#include <memory>
+
+#include <CL/sycl.hpp>
+
+#include "input.h"
+#include "modules.h"
+#include "output.h"
+#include "rawtodigi_oneapi.h"
+
+namespace {
+  constexpr int NLOOPS = 100;
+}
+
+namespace oneapi {
+
+  void exception_handler(cl::sycl::exception_list exceptions) {
+    for (auto const &exc_ptr : exceptions) {
+      try {
+        std::rethrow_exception(exc_ptr);
+      } catch (cl::sycl::exception const &e) {
+        std::cerr << "Caught asynchronous SYCL exception:\n" << e.what() << std::endl;
+      }
+    }
+  }
+
+  void analyze(cl::sycl::device device, Input const &input, Output &output, double &totaltime) {
+#if __SYCL_COMPILER_VERSION <= 20200118
+    // Intel oneAPI beta 4
+    cl::sycl::ordered_queue queue{device, exception_handler};
+#else
+    // Intel SYCL branch
+    cl::sycl::queue queue{device, exception_handler, cl::sycl::property::queue::in_order()};
+#endif
```

Comment:
Do I understand correctly that the upper branch is essentially  for the Codeplay's implementatoin, and the lower branch is for Intel's (which is more recent than the "beta 4")?

Reply:
No, they are both for the Intel branch:
  - the top branch is for the HEAD of the [sycl](https://github.com/intel/llvm/tree/sycl) branch
  - the bottom part is for the [beta04](https://software.intel.com/en-us/forums/intel-oneapi-base-toolkit/topic/848563) compiler, that was branched off some time in January

The Codeplay branch is less recent, but still able to work with the top branch.
Once beta05 is available (within a couple of months) I can remove the check.

Reply:
Ok. But now I'm confused: are the comments in the code correct, or your GH comment above ("top = HEAD, bottom = beta04")?

Reply:
The code is correct, I didnt check it before commenting here, so i May have swapped them...

Top is old (beta04), bottom is new (HEAD).

Reply:
Thanks!